kiwi/solver.h

```diff
@@ -150,6 +150,14 @@ class Solver
 		debug::dump( m_impl );
 	}
 
+	/* Dump a representation of the solver internals to a string.
+
+	*/
+	std::string dumps()
```

Comment:
Same thing here, make an overload for `::dump` which takes an `ostream` ref as second arg.